Fix polymorphic relationship handling in FixtureService

jsirish opened the pull request:
## Problem

When `has_one` relationships are configured as arrays (polymorphic relationships), the FixtureService was failing with a fatal error:

```
DataObject::get() cannot query non-subclass DataObject directly
```

This occurs when processing models with polymorphic relationships like:

```php
private static $has_one = [
    'Owner' => [
        'class' => DataObject::class,
        'type' => 'polymorphic'
    ]
];
```

## Solution

This PR adds proper handling for array-based (polymorphic) relationship configurations:

1. **Type checking**: Checks if `$relationClassName` is an array before processing
2. **Class extraction**: Extracts the `'class'` key from polymorphic relationship arrays
3. **Error handling**: Adds warnings for invalid polymorphic configurations
4. **Validation**: Ensures `field_classname_map` returns valid class name strings

## Changes

- Added `is_array()` check for `$relationClassName` in `processHasOneRelationships()`
- Extract `'class'` value from polymorphic relationship arrays
- Added warning messages for invalid polymorphic configurations
- Added validation for `field_classname_map` return values

## Testing

This fix has been tested with:
- Standard has_one relationships (unchanged behavior)
- Polymorphic has_one relationships defined as arrays
- Export tasks that previously failed with fatal errors

The fix allows the fixture export to continue gracefully, handling polymorphic relationships properly while maintaining backward compatibility with standard relationship definitions.

## Impact

- Fixes fatal errors when exporting fixtures containing polymorphic relationships
- Maintains backward compatibility with existing relationship configurations
- Improves error reporting for misconfigured polymorphic relationships

This fix enables the use of chrispenny/silverstripe-data-object-to-fixture with projects that use polymorphic relationships, particularly when working with advanced elemental blocks or other modules that leverage this SilverStripe feature.

jsirish committed:
Fix polymorphic relationship handling in FixtureService

When has_one relationships are configured as arrays (polymorphic relationships),
the code was failing with a fatal error trying to check if the array value existed
as a class configuration.

This fix:
- Checks if relationClassName is an array
- Extracts the 'class' key for polymorphic relationships
- Adds proper warning handling for invalid configurations
- Validates that field_classname_map returns valid class names

Fixes fatal error: DataObject::get() cannot query non-subclass DataObject directly
when processing polymorphic has_one relationships defined as arrays.

chrispenny commented:
Thanks so much, @jsirish! Very thorough!

On the face of it, this sounds good. I must confess that I'm a little short on spare time to test the changes at the moment, so I'm not too sure when I'll be able to get to this. Making a friendly ping to @edwilde in case anyone in his team happens to have a spare moment coming up where they could give this change a go.

As an aside: Where the heck did Copilot come from?! I don't remember enabling that... I'm not entirely sure I agree with its recommendation, but I'll mull it over.

Thanks again :)

jsirish committed:
Refactor: Extract polymorphic relationship handling and improve error messages

- Extract polymorphic relationship logic into dedicated handlePolymorphicRelationship() method
- Replace var_export with json_encode for cleaner warning output
- Add better error handling for unexpected relationship types
- Addresses review feedback from PR #32

jsirish commented:
Hey @chrispenny no problem! I'm using the module for a SS5 project and thought I'd submit the patch, though I'm fine running off my branch for a while too. Maybe Copilot kicked in cause I used Copilot to submit the PR? Anyway this module saved me hours of time by creating fixtures so happy to help!
